src/core/memory/memory.cc

```diff
@@ -15,13 +15,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#ifndef DISABLE_WARNINGS
 
+#ifdef USE_CUDA
```

Comment:
USE_CUDA is defined in the singa_config.h file, which is included in memory.h.
should put memory.h before the macro